### PR TITLE
sign: Change COSESign1 internal struct field names

### PR DESCRIPTION
**Pull request by puiterwijk**
This changes the COSESign1 structure to a named structure, which makes
the code easier to reason about.
To ensure we still serialize to the correct values (a sequence),
manually implement Serialize.

Signed-off-by: Patrick Uiterwijk <patrick@puiterwijk.org>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the Apache License Version 2.0, as specified in the LICENSE file of this repository.

### CODE REVIEW

**Review by petreeftime — APPROVED** [review]

**Comment by petreeftime** [comment]
I'll rebase  it once I get a second review.

**Review by andraprs — APPROVED** [review]

**Comment by petreeftime** [comment]
I've rebased and merged this commit.

**Pull request closed without merging**
